apiserver/facades/client/bundle/bundle.go

```diff
@@ -224,7 +225,16 @@ func (b *BundleAPI) fillBundleData(model description.Model) (*charm.BundleData,
 		} else {
 			ut := []string{}
 			for _, unit := range application.Units() {
-				ut = append(ut, unit.Machine().Id())
+				machineID := unit.Machine().Id()
```

Comment:
we should not be using the childID(). only in case if machine.ID() is 0/lxd/0. There can be other cases where the parent machine is different 1/lxd/0. I have also referred to the tests in machine.go in description to understand this. Also, if u see in gui they use the parentID only.

Reply:
Sorry, I meant to say machine.Parent().Id()

Reply:
ok. agree. I missed Parent().ID().